x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/delete_engine_modal.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+
+import { useActions, useValues } from 'kea';
+
+import { EuiConfirmModal } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { EnginesListLogic } from './engines_list_logic';
+
+export const DeleteEngineModal: React.FC = () => {
+  const { closeDeleteEngineModal, deleteEngine } = useActions(EnginesListLogic);
+  const {
+    deleteModalEngineName: engineName,
+    isDeleteModalVisible,
+    isDeleteLoading,
+  } = useValues(EnginesListLogic);
+
+  if (isDeleteModalVisible) {
+    return (
+      <EuiConfirmModal
+        title={i18n.translate('xpack.enterpriseSearch.content.engineList.deleteEngineModal.title', {
+          defaultMessage: 'Permanently delete this engine?',
+        })}
+        onCancel={() => {
+          closeDeleteEngineModal();
+        }}
+        onConfirm={() => {
+          deleteEngine({ engineName });
+        }}
+        cancelButtonText={i18n.translate(
+          'xpack.enterpriseSearch.content.engineList.deleteEngineModal.closeButton.title',
+          {
+            defaultMessage: 'Cancel',
```

Comment:
suggestion: use the [CANCEL_BUTTON_LABEL](https://github.com/elastic/kibana/blob/main/x-pack/plugins/enterprise_search/public/applications/shared/constants/actions.ts#L33-L36) constant maybe?

Reply:
Yes. but [CANCEL_BUTTON_LABEL](https://github.com/elastic/kibana/blob/main/x-pack/plugins/enterprise_search/public/applications/shared/constants/actions.ts#L33-L36) may have different translation string from the one's mentioned in this file ?

Reply:
The default here is `Cancel` which is the same. Re-using the constant helps us not have N "Cancel"s in the translation file. But we definitely repeat other translations and aren't very strict on no duplicates.

Reply:
We used to be very strict about duplicates and having constants outside of the components, but that really just led to more issues than it solved tbh.

Reply:
Just to confirm. Do we still need to replace current translation with [CANCEL_BUTTON_LABEL](https://github.com/elastic/kibana/blob/main/x-pack/plugins/enterprise_search/public/applications/shared/constants/actions.ts#L33-L36) ?